Babylon file export.js

```diff
@@ -50,308 +52,528 @@ var BabylonMesh = function(obj)
-                        var light = new BabylonLight(child, 3);
-                        lights.push(light);
-                        break;
-                    case 2:
-                        var light = new BabylonLight(child, 1);
-                        lights.push(light);
-                        break;
-                    case 4:
-                        var light = new BabylonLight(child, 2);
-                        lights.push(light);
-                        break;
-                    default:
-                        var light = new BabylonLight(child, 0);
-                        lights.push(light);
-                        break;
-                }
-                break;
-            case CAMERA:
-                var camera = new BabylonCamera(child);
-                cameras.push(camera);
-                break;
-            default:
-                var mesh = new BabylonMesh(child);
-                if(parentId)
-                    mesh.parentId = parentId;
-                parentId = mesh.id;
-                if(!mesh.notMesh)
-                {
-                    meshes.push(mesh);
-                }
-                break;
-        }
-        if(child.childCount() > 0)
-            getChildren(child, parentId);
+class BabylonMaterial {
+  constructor(material) {
+    let rootnode = material.rootNode();            // get the root node of the material
+    let textures = {};
+    for (let j = 0; j < rootnode.inputCount(); j++) {
+      let param = rootnode.inputAtIndex(j);
+      if (param.connected() !== true) continue;    // if the parameter is not connected, exit
+      let nodeID = param.connectedWithID();        // Get the ID of the node this texture is connected with
+      let node = material.nodeWithID(nodeID);      // get the node with ID from the material
+      if (node === null) continue;                 // skip if node is null
+      if (node.nodeType() !== 'image') continue;   // skip if node is not an image
+      Object.assign(textures, {
+        [param.name()]: getParameters(node)
+      });
+    }
+
+    let name = material.getParameter('name');
+    let diffuse = material.color();
+    let specular = material.specular();
+    let emission = material.emission();
+
+    Object.assign(this, {
+      name:              name,
+      id:                name,
+      tags:              MATERIAL_TAGS,
+      disableDepthWrite: MATERIAL_DISABLE_DEPTH_WRITE,
+      ambient:           new BabylonColor3(0, 0, 0).export(),
+      diffuse:           new BabylonColor3(diffuse.x, diffuse.y, diffuse.z).export(),
+      specular:          new BabylonColor3(specular.x, specular.y, specular.z).export(),
+      specularPower:     material.shininess(),
+      emissive:          new BabylonColor3(emission.x, emission.y, emission.z).export(),
+      alpha:             1.0,
+      backFaceCulling:   true,
+      wireframe:         false,
+      /* diffuseTexture:                   new BabylonTexture(),
```

Comment:
comment cleanup (unless this is temporary?)

Reply:
I'll remove it; it's mostly just commented code I derived from the Babylon docs; was helpful to me to see all exposed attributes.